OptimoveSDK/Sources/Classes/Optimobile/InApp/InAppPresenter.swift

```diff
@@ -88,29 +103,44 @@ class InAppPresenter : NSObject, WKScriptMessageHandler, WKNavigationDelegate{
             }
         }
 
+        let notShowingCurrentTickle = self.currentMessage != nil
+            && self.currentMessage!.id != (self.messageQueue[0] as! InAppMessage).id
+            && (self.messageQueue[0] as! InAppMessage).id == self.pendingTickleIds[0] as! Int64
+
+        let queueNotEmptyAndNotShowingAnything = self.currentMessage == nil && self.messageQueue.count > 0
+
+        let shouldShowSomething = notShowingCurrentTickle || queueNotEmptyAndNotShowingAnything
+
         messageQueueLock.signal()
-      
-        DispatchQueue.main.async {
-            self.initViews()
-            
-            if (self.currentMessage != nil
-                && self.currentMessage!.id != (self.messageQueue[0] as! InAppMessage).id
-                && (self.messageQueue[0] as! InAppMessage).id == self.pendingTickleIds[0] as! Int64) {
+
+        if (shouldShowSomething) {
```

Comment:
We change the logic here to always call presentFromQueue if:

- there's a pending tickle that differs from the current message being displayed, or
- there's not currently a message being shown but we have a non-empty queue